packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts

```diff
@@ -821,6 +829,191 @@ describe('Collection aggregations tab', function () {
+      expect(await stageContent.getText()).to.equal(`{
+  i: 5,
+}`);
+    });
+
+    it('runs pipeline in text mode when changed', async function () {
+      await browser.selectStageOperator(0, '$match');
+      await browser.setAceValue(Selectors.stageEditor(0), '{ i: 5 }');
+      await switchPipelineMode(browser, 'as-text');
+
+      await browser.setAceValue(
+        Selectors.AggregationAsTextEditor,
+        '[{$count: "count"}]'
+      );
+
+      const docsPreview = await browser.$(
+        Selectors.AggregationAsTextPreviewDocument
+      );
+      await docsPreview.waitForDisplayed();
+      const text = (await docsPreview.getText())
+        .replace(/\n/g, ' ')
+        .replace(/\s+?:/g, ':')
+        .replace(/\s+/g, ' ');
+      expect(text).to.contain('count: 1000');
+    });
+
+    it('previews $out stage', async function () {
+      await browser.selectStageOperator(0, '$match');
+      await browser.setAceValue(Selectors.stageEditor(0), '{ i: 5 }');
+      await switchPipelineMode(browser, 'as-text');
+
+      await browser.setAceValue(
+        Selectors.AggregationAsTextEditor,
+        '[{$out: "somewhere"}]'
+      );
+
+      const preview = await browser.$(Selectors.AggregationAsTextPreviewOut);
+      await preview.waitForDisplayed();
+      const text = await preview.getText();
+      expect(text).to.contain(
+        'The $out operator will cause the pipeline to persist the results to the specified location'
+      );
+    });
+
+    it('previews $merge stage', async function () {
+      await browser.selectStageOperator(0, '$match');
+      await browser.setAceValue(Selectors.stageEditor(0), '{ i: 5 }');
+      await switchPipelineMode(browser, 'as-text');
+
+      await browser.setAceValue(
+        Selectors.AggregationAsTextEditor,
+        '[{$merge: "somewhere"}]'
+      );
+
+      const preview = await browser.$(Selectors.AggregationAsTextPreviewMerge);
+      await preview.waitForDisplayed();
+      const text = await preview.getText();
+      expect(text).to.contain(
+        'The $merge operator will cause the pipeline to persist the results to the specified location'
+      );
+    });
+
+    it('previews atlas operators - $search', async function () {
+      await browser.selectStageOperator(0, '$match');
+      await browser.setAceValue(Selectors.stageEditor(0), '{ i: 5 }');
+      await switchPipelineMode(browser, 'as-text');
+
+      await browser.setAceValue(
+        Selectors.AggregationAsTextEditor,
+        '[{$search: {}}]'
+      );
+
+      const preview = await browser.$(
+        Selectors.AggregationAsTextPreviewAtlasOperator
+      );
+      await preview.waitForDisplayed();
+      expect(await preview.getText()).to.include(
+        'The $search stage is only available with MongoDB Atlas'
+      );
+
+      const errors = await browser.$(Selectors.AggregationAsTextErrorContainer);
+      expect(await errors.getText()).to.include(
+        "Unrecognized pipeline stage name: '$search'"
```

Comment:
Seems like these tests don’t [always result in the same message](https://spruce.mongodb.com/task/10gen_compass_main_ubuntu_test_packaged_app_44x_enterprise_patch_811f47910a772a6df613a9f2590388a19adcb976_63760c363e8e86331034a980_22_11_17_10_26_00/logs?execution=0):

```
[2022/11/17 10:42:04.061]   2 failing
[2022/11/17 10:42:04.062]   1) Collection aggregations tab
[2022/11/17 10:42:04.062]        aggregation builder in text mode
[2022/11/17 10:42:04.062]          previews atlas operators - $search:
[2022/11/17 10:42:04.062]      AssertionError: expected '$search not enabled! Enable Search by…' to include 'Unrecognized pipeline stage name: \'$…'
[2022/11/17 10:42:04.062]       at Context.<anonymous> (tests/collection-aggregations-tab.test.ts:945:41)
[2022/11/17 10:42:04.062]       at runMicrotasks (<anonymous>)
[2022/11/17 10:42:04.062]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
[2022/11/17 10:42:04.062]   2) Collection aggregations tab
[2022/11/17 10:42:04.062]        aggregation builder in text mode
[2022/11/17 10:42:04.062]          previews atlas operators - $searchMeta:
[2022/11/17 10:42:04.062]      AssertionError: expected '$search not enabled! Enable Search by…' to include 'Unrecognized pipeline stage name: \'$…'
[2022/11/17 10:42:04.062]       at Context.<anonymous> (tests/collection-aggregations-tab.test.ts:969:41)
[2022/11/17 10:42:04.062]       at runMicrotasks (<anonymous>)
[2022/11/17 10:42:04.062]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Reply:
fixed! thanks!